In [2]:
import numpy as np
import pandas as pd
import tensorflow as tf
import matplotlib.pyplot as plt

In [ ]:
n = 1000
celsius = np.random.randint(-n,n)
res = np.random.randn(n) 
farenheit = (9/5)*celsius +32 + res

In [ ]:
model = tf.keras.Sequential()
model

In [ ]:
model.add(tf.keras.Input(shape=(1)))
model.add(tf.keras.layers.Dense(1))

In [ ]:
layer = tf.keras.layers.Dense(units = 1, input_shape=[1])
model = tf.keras.Sequential([capa1])

NameError: ignored

In [ ]:
model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

In [ ]:
model.fit(celsius, y_train, epochs=5)

model.evaluate(x_test,  y_test, verbose=2)

#Tensor Flow beginner example

In [3]:
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

11490434/11490434 [==============================] - 0s 0us/step


In [9]:
x_train.shape

(60000, 28, 28)

In [10]:
y_train.shape

(60000,)

In [6]:
(x_train.shape, y_train.shape)

((60000, 28, 28), (60000,))

Construye un modelo `tf.keras.Sequential`  apilando capas. Escoge un optimizador y una funcion de perdida para el entrenamiento de tu modelo:

In [7]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')#Funcion de activacion de normalizacion
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

Entrena y evalua el modelo:

In [8]:
model.fit(x_train, y_train, epochs=10)

model.evaluate(x_test,  y_test, verbose=2)

Epoch 1/10
1875/1875 [==============================] - 10s 4ms/step - loss: 0.2483 - accuracy: 0.9253
Epoch 2/10
1875/1875 [==============================] - 13s 7ms/step - loss: 0.1069 - accuracy: 0.9675
Epoch 3/10
1875/1875 [==============================] - 12s 7ms/step - loss: 0.0773 - accuracy: 0.9762
Epoch 4/10
1875/1875 [==============================] - 9s 5ms/step - loss: 0.0614 - accuracy: 0.9817
Epoch 5/10
1875/1875 [==============================] - 11s 6ms/step - loss: 0.0502 - accuracy: 0.9842
Epoch 6/10
1875/1875 [==============================] - 10s 5ms/step - loss: 0.0436 - accuracy: 0.9863
Epoch 7/10
1875/1875 [==============================] - 9s 5ms/step - loss: 0.0360 - accuracy: 0.9885
Epoch 8/10
1875/1875 [==============================] - 9s 5ms/step - loss: 0.0320 - accuracy: 0.9896
Epoch 9/10
1875/1875 [==============================] - 7s 4ms/step - loss: 0.0291 - accuracy: 0.9906
Epoch 10/10
1875/1875 [==============================] - 9s 5ms/step - loss: 

[0.09621668606996536, 0.9768999814987183]

El model de clasificacion de images fue entrenado y alcanzo una exactitud de ~98% en este conjunto de datos. Para aprender mas, lee los [tutoriales de TensorFlow].